src/xmagics.hpp

```diff
@@ -31,19 +32,23 @@ namespace xeus
     {
     };
 
-    class xmagics_manager 
+    class xmagics_manager: public xpreamble
     {
     public:
 
-        xmagics_manager(){};
+        xmagics_manager()
+        {
+            xpreamble::set_pattern(R"(^(?:\%{2}|\%)(\w+))");
+        }
 
         template<typename xmagic_type>
-        void register_magic(const std::string& magic_name, std::shared_ptr<xmagic_type> magic)
+        void register_magic(const std::string& magic_name, xmagic_type magic)
         {
+            auto shared = std::make_shared<xmagic_type>(magic);
```

Comment:
Why to we need a shared pointer?

Reply:
we need a shared_ptr because a magic function could be a magic_cell and a magic_line.